packages/puppeteer-core/src/common/bidi/Input.ts

```diff
@@ -0,0 +1,326 @@
+      context: this.#context.id,
+      actions: [
+        {
+          type: Bidi.Input.SourceActionsType.Pointer,
+          id: 'main',
+          actions,
+        },
+      ],
+    });
+  }
+
+  override async wheel(
+    options: Readonly<MouseWheelOptions> = {}
+  ): Promise<void> {
+    await this.#context.connection.send('input.performActions', {
+      context: this.#context.id,
+      actions: [
+        {
+          type: Bidi.Input.SourceActionsType.Wheel,
+          id: 'main',
+          actions: [
+            {
+              type: Bidi.Input.ActionType.Scroll,
+              ...(this.#lastMovePoint ?? {
+                x: 0,
+                y: 0,
+              }),
+              deltaX: options.deltaX ?? 0,
+              deltaY: options.deltaY ?? 0,
+            },
+          ],
+        },
+      ],
+    });
+  }
+}
+
+/**
+ * @internal
+ */
+export class Touchscreen extends BaseTouchscreen {
+  #context: BrowsingContext;
+
+  /**
+   * @internal
+   */
+  constructor(context: BrowsingContext) {
+    super();
+    this.#context = context;
+  }
+
+  override async tap(
+    x: number,
+    y: number,
+    options: BidiTouchMoveOptions = {}
+  ): Promise<void> {
+    await this.touchStart(x, y, options);
+    await this.touchEnd();
+  }
+
+  override async touchStart(
+    x: number,
+    y: number,
+    options: BidiTouchMoveOptions = {}
+  ): Promise<void> {
+    await this.#context.connection.send('input.performActions', {
+      context: this.#context.id,
+      actions: [
+        {
+          type: Bidi.Input.SourceActionsType.Pointer,
+          id: 'main',
+          parameters: {
+            pointerType: Bidi.Input.PointerType.Touch,
+          },
+          actions: [
+            {
+              type: Bidi.Input.ActionType.PointerMove,
+              x,
+              y,
+              origin: options.origin,
+            },
+            {
+              type: Bidi.Input.ActionType.PointerDown,
+              button: 0,
+            },
+          ],
+        },
+      ],
+    });
+  }
+
+  override async touchMove(
+    x: number,
+    y: number,
+    options: BidiTouchMoveOptions = {}
+  ): Promise<void> {
+    await this.#context.connection.send('input.performActions', {
+      context: this.#context.id,
+      actions: [
+        {
+          type: Bidi.Input.SourceActionsType.Pointer,
+          id: 'main',
+          parameters: {
+            pointerType: Bidi.Input.PointerType.Touch,
+          },
+          actions: [
+            {
+              type: Bidi.Input.ActionType.PointerMove,
+              x,
+              y,
+              origin: options.origin,
+            },
+          ],
+        },
+      ],
+    });
+  }
+
+  override async touchEnd(): Promise<void> {
+    await this.#context.connection.send('input.performActions', {
+      context: this.#context.id,
+      actions: [
+        {
+          type: Bidi.Input.SourceActionsType.Pointer,
+          id: 'main',
```

Comment:
If I understand correctly BiDi allows for multiple input form the the same group (You can have 2 keyboards, both sending separate comands).
We should extract this `main` somewhere as on Puppeteer side we want to keep using only one for now.